articles/architecture-scenarios/application/mobile-api.md

```diff
@@ -4,35 +4,290 @@ title: Mobile + API
+- __token_type__: A string containing the type of token, this will always be a Bearer token.
+- __expires_in__: The amount of seconds until the `access_token` expires.
+
+You will need to store the above credentials in local storage for use in calling your API and retrieving the user profile.
+
+::: note
+[See the implementation in Android.](/architecture-scenarios/application/mobile-api/mobile-implementation-android#store-credentials)
+:::
+
+#### Get the User Profile
+
+To retrieve the [User Profile](/api/authentication?http#user-profile), your mobile application can decode the [ID Token](/tokens/id-token) using one of the [JWT libraries](https://jwt.io/#libraries-io). This is done by [verifying the signature](/tokens/id-token#verify-the-signature) and [validating the claims](/tokens/id-token#validate-the-claims) of the token. After validating the ID token, you can access its payload containing the user information:
+
+```json
+{
+  "email_verified": false,
+  "email": "test.account@userinfo.com",
+  "clientID": "q2hnj2iu...",
+  "updated_at": "2016-12-05T15:15:40.545Z",
+  "name": "test.account@userinfo.com",
+  "picture": "https://s.gravatar.com/avatar/dummy.png",
+  "user_id": "auth0|58454...",
+  "nickname": "test.account",
+  "created_at": "2016-12-05T11:16:59.640Z",
+  "sub": "auth0|58454..."
+}
+```
+
+::: note
+[See the implementation in Android.](/architecture-scenarios/application/mobile-api/mobile-implementation-android#3-get-the-user-profile)
+:::
+
+#### Call the API
+
+To access secured resources from your API, the authenticated user's `access_token` needs to be included in requests that are sent to it. This is accomplished by sending the `access_token` in an `Authorization` header using the `Bearer` scheme.
+
 ::: note
-This architecture scenario is under construction and will be updated soon.
+[See the implementation in Android.](/architecture-scenarios/application/mobile-api/mobile-implementation-android#4-call-the-api)
 :::
 
-![Mobile + API Flow](/media/articles/architecture-scenarios/mobile-api.png)
+#### Renew the Token
+
+To refresh your `access_token`, perform a `POST` request to the `/oauth/token` endpoint using the `refresh_token` from your authorization result.
+
+A [Refresh Token](/tokens/refresh-token/current) will only be present if you included the `offline_access` scope in the previous authorization request and  enabled __Allow Offline Access__ for your API in the Dashboard.
 
-In this scenario you have a mobile application ("Client") which talks to an API ("Resource Server"). The application will use **OpenID Connect** with the **Authorization Code Grant using Proof Key for Code Exchange (PKCE)** to authenticate users. Note that this flow can only be used for Clients whose type is **Native** in the [Dashboard](${manage_url}).
+Your request should include:
 
-When a user logs in, Auth0 will return to the application an `access_token`, an `id_token`, and optionally a `refresh_token`:
+Parameter | Description
```

Comment:
In all the other cases first you have the snippet, and then you explain the params (which makes more sense). Move this table after the code snippet